### PR TITLE
New rule: no-extended-unicode-identifiers

### PR DESCRIPTION
**Pull request by catrope**
This rule prohibits non-BMP characters from being used in identifiers.
It can be used to write code that has to run in browsers that support
all other ES6 language features except this one (Safari and iOS versions
11-13).

See https://phabricator.wikimedia.org/T300666

### CODE REVIEW

**Review by jdforrester — DISMISSED** [review]
LGTM. Does anyone else have a view or should we just merge?

**Comment by DannyS712** [comment]
LGTM, but should this also handle vue data and props that likewise shouldn't use these characters?

File: src/rules/no-extended-unicode-identifiers.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+module.exports = {
+	meta: {
+		type: 'problem',
+		docs: {
+			description: 'Prohibits non-BMP characters in identifiers, for Safari 11-13 compatibility'
```
Comment (Krinkle) [inline src/rules/no-extended-unicode-identifiers.js:7]: Nit: Can you add a full stop at the end of both of these? (I'd do an edit, but it needs the markdown file to be regenerated and I'd rather not make a mess of the branch.)

**Review by Krinkle — APPROVED** [review]

**Comment by catrope** [comment]
Sorry, accidentally rebased + force-pushed instead of adding another commit, so seeing the diff is harder. But all I did is add full stops per @Krinkle's review

**Review by Krinkle — APPROVED** [review]

**Pull request merged**